exotic.py

```diff
@@ -1590,163 +1711,177 @@ def parse_args():
-
-            # apply cals correction if applicable
-            if darksBool:
-                print("Dark subtracting images.")
-                sortedallImageData = sortedallImageData - generalDark
-            elif biasesBool:
-                print("Bias-correcting images.")
-                sortedallImageData = sortedallImageData - generalBias
-            else:
-                pass
-
-            if flatsBool:
-                print("Flattening images.")
-                sortedallImageData = sortedallImageData / generalFlat
-
-            pathSolve = saveDirectory + 'first_file.fits'
-            try:
-                os.remove(pathSolve)
-            except OSError:
-                pass
-            convertToFITS = fits.PrimaryHDU(data=sortedallImageData[0])
-            convertToFITS.writeto(pathSolve)
-            wcsFile = check_wcs(pathSolve, saveDirectory)
-            if wcsFile:
-                hdulWCS = fits.open(name=wcsFile, memmap=False, cache=False, lazy_load_hdus=False)  # opens the fits file
-                rafile, decfile = get_radec(hdulWCS)
-
-            print("\nAligning your images from .FITS. Please wait.")
-            done = False
-            t = threading.Thread(target=animate, daemon=True)
-            t.start()
-            sortedallImageData, unalignedBoolList, boollist = image_alignment(sortedallImageData)
-            done = True
-            print('\n\nImages Aligned.')
+                    os.remove(pathSolve)
+                except OSError:
+                    pass
+                convertToFITS = fits.PrimaryHDU(data=sortedallImageData[0])
+                convertToFITS.writeto(pathSolve)
+                wcsFile = check_wcs(pathSolve, saveDirectory)
+
+                # Check pixel coordinates by converting to WCS. If not correct, loop over again
+                if wcsFile:
+                    hdulWCS = fits.open(name=wcsFile, memmap=False, cache=False, lazy_load_hdus=False)  # opens the fits file
+
+                    # Save previously entered x and y pixel coordinates before checking against plate solution
+                    saveUIprevTPX, saveUIprevTPY = UIprevTPX, UIprevTPY
+                    UIprevTPX, UIprevTPY = check_targetpixelwcs(UIprevTPX, UIprevTPY, pDict['ra'],
+                                                                pDict['dec'], hdulWCS)
+
+                    # If the coordinates were not changed, do not loop over again
+                    if UIprevTPX != saveUIprevTPX or UIprevTPY != saveUIprevTPY:
+                        continue
+
+                # Image Alignment
+                print("\nAligning your images from .FITS. Please wait.")
+                done = False
+                t = threading.Thread(target=animate, daemon=True)
+                t.start()
+                sortedallImageData, unalignedBoolList, boollist = image_alignment(sortedallImageData)
+                done = True
+                print('\n\nImages Aligned.')
+                break
```

Comment:
The idea behind tracking the target pixel coordinates is to do it by these steps: time sort, plate solution, pixel coordinates, then image alignment. I did image alignment last because I wanted to check the pixel coordinates to save time for the user having to do image alignment twice.

Reply:
Right, but if a user defines their xy pixels on the first image but the 10th image is chosen as the one that all others are aligned, there is a chance (albeit likely small...hopefully) that the image will shift enough on the focal plane that the user will select the "wrong" star. But perhaps your checking code will help prevent this.

For now, I would tell the user what image we are using as the "first" one (both the image number and the full filename)- that way they can open that FITS file later to confirm the star's xy pixel coordinates.

Reply:
Done in the newest push. Created a file with the original name and gives the image number.